pages/src/TodoList.js

```diff
@@ -2,123 +2,38 @@ import ImmutablePropTypes from 'react-immutable-proptypes'
 import PropTypes from 'prop-types'
 import React, { Component } from 'react'
 import { createSelector } from 'reselect'
-
-export default class TodoList extends Component {
-  static propTypes = {
-    onCreateTodo: PropTypes.func.isRequired,
-    onEditTodo: PropTypes.func.isRequired,
-    todos:ImmutablePropTypes.mapOf(
-      PropTypes.shape({
-        completed: PropTypes.bool.isRequired,
-        label: PropTypes.string.isRequired,
-        id: PropTypes.string.isRequired
-      }).isRequired,
-      PropTypes.string.isRequired
-    )
-  }
-
-  state = {
-    display: 'all',
-    newTodoLabel: ''
-  }
-
-  _createTodo = event => {
-    event.preventDefault()
-
-    const newTodoLabel = this.state.newTodoLabel.trim()
-    if (newTodoLabel === '') {
-      return
-    }
-
-    this.setState({ newTodoLabel: '' })
-
-    // all items of collection should have a unique identifier
-    const id = Math.random().toString(36).slice(2)
-    const todo = {
-      completed: false,
-      id,
-      label: newTodoLabel
-    }
-
-    this.props.onCreateTodo(todo)
-  }
-
-  // - what's a selector?
-  // - what's a memoized selector?
-  //
-  // see https://github.com/reactjs/reselect
-  _getVisibleTodos = createSelector(
-    state => state.display,
-    (_, props) => props.todos,
-    (display, todos) => {
-      if (display === 'all') {
-        return todos.valueSeq()
-      }
-
-      return todos.valueSeq().filter(
-        display === 'active'
-          ? t => !t.completed
-          : t => t.completed
+import { injectState } from "freactal";
+import { Map } from 'immutable'
+
+const TodoList = injectState(({ state, effects }) => {
+  
+      const onNewTodoLabelChange = ev => effects.setNewTodoLabelChange(ev.target.value)
+      const onCreateTodo = ev => { ev.preventDefault()
```

Comment:
Avoid creating event handlers in the render function, perhaps they could be defined as effects?

Reply:
when I sended the event to an effect and tried to execute : event.preventDefault() it didn't work.. I tried hard to do it but I couldn't that's why i did it this way

Reply:
I don't know how to send the event directly like so : 
<Input onChange={ effects.onNewTodoLabelChange(event) } </event>
It's doen't work on JSX